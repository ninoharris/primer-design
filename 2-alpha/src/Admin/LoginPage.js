import React from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { startLogin } from '../actions/auth'

export const LoginPage = ({
  startLogin
}) => {
  return (
    <div className="Login-Page">
      <div className="container-fluid">
        <div className="row Login-Page-Header">
          Primer designer admin hub
        </div>
        <div className="row">
          <div className="col-4">
            Log in to admin section
            <button onClick={startLogin} className="btn btn-primary">Log in</button>
          </div>
        </div>
      </div>
    </div>
  )
}
LoginPage.propTypes = {
  startLogin: PropTypes.func.isRequired,
}

const mapDispatchToProps = (dispatch) => {
  return {
    startLogin: () => dispatch(startLogin)
  }
}

// export default connect(null, { startLogin })(LoginPage)
export default connect(null, mapDispatchToProps)(LoginPage)