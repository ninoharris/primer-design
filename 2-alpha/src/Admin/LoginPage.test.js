import React from 'react'
import { LoginPage } from './LoginPage'
import { shallow } from 'enzyme'

test('Snapshot of LoginPage renders properly', () => {
  const startLogin = jest.fn()
  const wrapper = shallow(<LoginPage startLogin={startLogin} />)
  expect(wrapper).toMatchSnapshot()
  wrapper.find('.Login-Button').simulate('click')
  expect(startLogin).toHaveBeenCalledTimes(1)
})