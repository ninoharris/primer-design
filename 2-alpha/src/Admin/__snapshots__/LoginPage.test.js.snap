// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of LoginPage renders properly 1`] = `
<div
  className="Login-Page"
>
  <div
    className="container-fluid"
  >
    <div
      className="row Login-Page-Header"
    >
      Primer designer admin hub
    </div>
    <div
      className="row"
    >
      <div
        className="col-4"
      >
        Log in to admin section
        <button
          className="btn btn-primary"
          onClick={[MockFunction]}
        >
          Log in
        </button>
      </div>
    </div>
  </div>
</div>
`;
